static/js/tools/choropleth/choropleth.tsx

```diff
@@ -361,6 +389,70 @@ class ChoroplethMap extends Component<PropsType, StateType> {
           .attr("font-weight", "bold")
           .text(title)
       );
+
+    // Place the date dropdown on the screen.
+    const datePickerComponent = this.datePicker();
+    ReactDOM.render(datePickerComponent, document.getElementById("date"));
+  };
+
+  /**
+   *
+   * Changes the date and updates the values on the state.
+   * This causes a re-render that makes the map be updated
+   * to reflect the new data.
+   * @param event: an 'onChange' event.
+   */
+  public changeDate = (event: { target: { value: string } }): void => {
+    // Get the newly selected date as a string.
+    const newDate: string = event.target.value;
+    console.log(newDate);
+    const data = this.state["data"];
+    // Re-update this.values to only include data for that given date.
+    const geoIdToValue = this.filterByDate(data, newDate);
+    // Store the new date and the new values.
+    this.state.values = geoIdToValue;
+    this.setState({ date: newDate });
+
+    // Re-render component and map.
+    this.renderGeoMap();
```

Comment:
shouldn't updateGeoValues get called before renderGeoMap?

Reply:
Documentation for updateGeoValues says:

   ```* Requires geoJson map to be rendered and state values to be set.```

renderGeoMap() renders a blank map and then updateGeoValues draws the colors.
On all other instances, renderGeoMap() is called before updateGeoValues().

I have renamed renderGeoMap to drawBlankGeoMap.
and updateGeoValues to addColorToGeoMap.

Can change it to any other name if you'd like.

Reply:
that sounds good